Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,293 @@
+using Cilsil.Utils;
+using Cilsil.Sil;
+using Cilsil.Sil.Types;
+using Mono.Cecil.Cil;
+using Mono.Cecil;
+using Cilsil.Sil.Instructions;
+using Cilsil.Sil.Expressions;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Cilsil.Cil.Parsers
+{
+    internal class ExceptionParser : InstructionParser
+    {
+        protected override bool ParseCilInstructionInternal(Instruction instruction,
+                                                            ProgramState state)
+        {
+            var endBlockOffset = state.ExceptionBlockStartToEndOffsets[state.CurrentInstruction.Offset];
+            
+            /*node1 preds: succs:2 3 exn: Instructions
+            n$47=*&amp;return:void [line 55];   # load return
+            *&amp;return:void=null [line 55];
+            n$48=_fun___unwrap_exception(n$47:void) [line 55];*/
+
+            // Create exception handler node.
+            var exceptionHandlerNode = new StatementNode(state.CurrentLocation,
+                                                         StatementNode.StatementNodeKind.ExceptionHandler,
+                                                         state.ProcDesc);
+            // Load expression from stack. 
+            (var expression, var expressionType) = state.Pop(); 
+            var identifier = state.GetIdentifier(Identifier.IdentKind.Normal);
+            exceptionHandlerNode.Instructions.Add(
+                    new Load(identifier,
+                             expression,
+                             expressionType,
+                             state.CurrentLocation));
+
+            // Clear the loaded expression to null.
```

Comment:
Assigns the return value to null, to reflect the expected return value in the event of an exception being encountered.

Reply:
Done